lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -136,10 +136,20 @@ namespace {
             if (requiresUnsharedVertices) {
                 const VtIntArray& faceVertexIndices = topology.GetFaceVertexIndices();
                 if (numVertices == faceVertexIndices.size()) {
+                    unsigned int dataSize = primvarData.size();
                     for (size_t v = 0; v < numVertices; v++) {
                         SRC_TYPE* pointer = reinterpret_cast<SRC_TYPE*>(
                             reinterpret_cast<float*>(&vertexBuffer[v]) + channelOffset);
-                        *pointer = primvarData[faceVertexIndices[v]];
```

Comment:
We have checks comparing the size of primvarData and faceVertexIndices arrays, but we weren't checking the actual value that we are getting out of faceVertexIndices when using it as an index to the primvarData array.

Reply:
It sounds like a corrupted scene or a bug in upstream software producing USD data causing our inefficiency (due to more checking).